sklearn/utils/_array_api.py

```diff
@@ -105,6 +105,25 @@ def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
             yield array_namespace, None, None
 
 
+def _get_namespace_device_dtype_ids(param):
+    """Get pytest parametrization IDs for `yield_namespace_device_dtype_combinations`"""
+    # See #31042 for details
+    try:
+        import array_api_strict
+    except ImportError:
+        return f"{param}"
```

Comment:
I would return None here to keep the default naming strategy in most cases.

IIUC the [pytest doc](https://docs.pytest.org/en/stable/example/parametrize.html#different-options-for-test-ids) returning `None` use the general strategy.